x-pack/plugins/apm/public/components/fleet_integration/apm_policy_form/settings_form/index.tsx

```diff
@@ -70,7 +70,12 @@ function FormRow({
               </EuiText>
             }
           >
-            <FormRowSetting row={row} onChange={onChange} value={value} />
+            <FormRowSetting
+              row={row}
+              onChange={onChange}
+              value={value}
+              isDisabled={row.readOnly || frozen}
```

Comment:
What's the difference between frozen and readOnly?

Reply:
`readyOnly` is set on the field's definition settings based on if it is on Cloud or not and `frozen` is the property that comes with the policy. Based on @simitt comment [below](https://github.com/elastic/kibana/pull/126497#issuecomment-1055126527)  `frozen` must be enough for the logic to disable the fields, so I removed the `readOnly` and I'll make some test to make sure if that is the case.

Reply:
Okay, that makes sense. We should probably not have called it `readOnly` but instead kept it as `isCloud`. But that doesn't matter now. Good to see it simplified.